tools/__tasks__/validate-head/javascript.js

```diff
@@ -0,0 +1,27 @@
+const execa = require('execa');
+const getChangedFiles = require('../lib/get-changed-files');
+
+module.exports = {
+    description: 'Lint changed JS',
+    task: [
+        {
+            description: 'Lint changed JS',
+            task: () =>
+                getChangedFiles().then(files => {
+                    const jsFiles = files.filter(
+                        file =>
+                            file.endsWith('.js') ||
+                            file === 'git-hooks/pre-push'
+                    );
+                    const config = ['--quiet', '--color'];
+
+                    return execa('eslint', jsFiles.concat(config));
```

Comment:
this will lint any files with un-pushed, committed changes in their current state, is that right? 

i think you can lint the actual final version of the code you're pushing by doing something like:
```
git show HEAD:changed_file_path | eslint --stdin --stdin-filename changed_file_path
```